albatross/cereal/eigen.h

```diff
@@ -20,66 +20,82 @@
 
 namespace cereal {
 
-template <class Archive>
-inline void save(Archive &archive, const Eigen::VectorXd &v) {
-  size_type s = static_cast<size_type>(v.size());
-  archive(cereal::make_size_tag(s));
-  for (size_type i = 0; i < s; i++) {
-    archive(v[i]);
+/*
+ * The subsequent save/load methods catch the serialization methods
+ * for arbitrary Eigen::Matrix* types.  The general idea is that each
+ * row is saved as a Eigen::Vector.
+ */
+template <class Archive, typename _Scalar, int _Rows, int _Cols>
+inline void save(Archive &archive,
+                 const Eigen::Matrix<_Scalar, _Rows, _Cols> &v) {
+  size_type rows = static_cast<size_type>(v.rows());
+  archive(cereal::make_size_tag(rows));
+  for (size_type i = 0; i < rows; i++) {
+    Eigen::Matrix<_Scalar, _Cols, 1> row = v.row(i);
+    archive(row);
   }
 };
 
-template <class Archive>
-inline void load(Archive &archive, Eigen::VectorXd &v) {
-  size_type s = static_cast<size_type>(v.size());
-  archive(cereal::make_size_tag(s));
-  v.resize(s);
-  for (size_type i = 0; i < s; i++) {
-    archive(v[i]);
+template <class Archive, typename _Scalar, int _Rows, int _Cols>
+inline void load(Archive &archive, Eigen::Matrix<_Scalar, _Rows, _Cols> &v) {
+  size_type rows;
+  archive(cereal::make_size_tag(rows));
+  /*
+   * In order to determine the size of a matrix, we have to first determine
```

Comment:
Would it be easier to serialize the dimensions instead of needing to scan through?

Reply:
good point, I'm fine with whichever you think is easiest.

Reply:
alright, I changed it around so the rows and cols are stored explicitly.  Shouldn't add much overhead in storage and makes the code easier to read (I think).